tests/integration/16.IPsec.Tests.ps1

```diff
@@ -0,0 +1,85 @@
+#Each functional area in NSX should have a separate test file.
+
+#Try to group related tests in contexts.  Especially ones that rely on configuration done in previous tests
+#Try to make tests as standalone as possible, but generally round trips to the API are expensive, so bear in mind
+#the time spent recreating configuration created in previous tests just for the sake of keeping test isolation.
+
+#Try to put all non test related setup and tear down in the BeforeAll and AfterAll sections.  ]
+#If a failure in here occurs, the Describe block is not executed.
+
+#########################
+#Do not remove this - we need to ensure connection setup and module deps preload have occured.
+If ( -not $PNSXTestVC ) {
+    Throw "Tests must be invoked via Start-Test function from the Test module.  Import the Test module and run Start-Test"
+}
+
+Describe "Edge IPsec" {
+
+    BeforeAll {
+
+        #BeforeAll block runs _once_ at invocation regardless of number of tests/contexts/describes.
+        #We load the mod and establish connection to NSX Manager here.
+
+        #Put any setup tasks in here that are required to perform your tests.  Typical defaults:
+        import-module $pnsxmodule
+        $script:DefaultNsxConnection = Connect-NsxServer -vCenterServer $PNSXTestVC -NsxServerHint $PNSXTestNSX -Credential $PNSXTestDefMgrCred -ViWarningAction "Ignore"
+        $script:cl = get-cluster | select -first 1
+        write-warning "Using cluster $cl for clustery stuff"
+        $script:ds = $cl | get-datastore | select -first 1
+        write-warning "Using datastore $ds for datastorey stuff"
+
+        #Put any script scope variables you need to reference in your tests.
+        #For naming items that will be created in NSX, use a unique prefix
+        #pester_<testabbreviation>_<objecttype><uid>.  example:
+        $script:ipsecedge1name = "pester-ipsec-edge1"
+        $script:ipsecedge1ipuplink = "1.1.1.1"
+        $script:password = "VMware1!VMware1!"
+        $script:ipsecuplinklsname = "pester_ipsec_uplink_ls"
+
+        #Create Logical Switch
+        $script:ipsecuplinkls = Get-NsxTransportZone -LocalOnly | select -first 1 | New-NsxLogicalSwitch $ipsecuplinklsname
+
+        #Create Edge Interface
+        $vnic0 = New-NsxEdgeInterfaceSpec -index 0 -Type uplink -Name "vNic0" -ConnectedTo $ipsecuplinkls -PrimaryAddress $ipsecedge1ipuplink -SubnetPrefixLength 24
+
+        #Create Edge
+        $script:sslEdge = New-NsxEdge -Name $ipsecedge1name -Interface $vnic0 -Cluster $cl -Datastore $ds -password $password -enablessh -hostname $ipsecedge1name
+
+    }
+
+    Context "ipsec" {
+
+        #Group related tests together.
+
+        it "IPsec is Disabled (by default)" {
+
+            #By default IPsec on Edge is disabled (and need to add new "Site" for enable) but no yet PowerNSX cmdlet for this
+            #Check if IPsec is disable
+            (Get-NsxEdge $ipsecedge1name).features.ipsec.enabled | should be false
```

Comment:
What are you testing here?  If you are trying to validate that the env is _read_ to test, then that should not be in a test itself, but in the VeforeEach/All block.  Your tests should concentrate on 'When I issues this command, does the response indicate that what I asked to happen, actually did.  So your test should be something like it "can enable IPSet"  and contain the code to both enable it, and test that it is enabled by examining the response (or the result of a new get)

If this is just a place holder and you arent finished yet, maybe dont submit the PR until you have a set of commits to introduce new functionality and the tests that go with them.

Reply:
at origine, i will enable IPsec but after some test, we need to add IPsec Site for this.. and there is no yet code...

i will plan after to update #354 commit to use this test...
i will prefer to merge #354 before start to work on other stuff...

Reply:
Lets keep it simple.  You need to add tests in #354, that test the functionality being added in #354.  When you have done that, I can merge it.  If you are planning on adding functionality being tested in this PR, then add it and update the tests to test it and let me know when done so I can take a look.

Reply:
I want to add a initial test on this PR and after update this file (and #354) to add specific IPsec Stats tests...
and it is to known if it is good start...

Reply:
Well - refer to my first comment, but yes, the expected approach is to copy the template file as you have done and then populate with suitable tests.  When you have done this, let me know and I will comment.  :)